parser-typechecker/src/Unison/Codebase/FileCodebase/Common.hs

```diff
@@ -280,13 +281,15 @@ codebaseExists root =
   and <$> traverse doesDirectoryExist (minimalCodebaseStructure root)
 
 -- | load a branch w/ children from a FileCodebase
-branchFromFiles :: MonadIO m => CodebasePath -> Branch.Hash -> m (Maybe (Branch m))
-branchFromFiles rootDir h = do
+branchFromFiles :: MonadIO m => Branch.Cache m -> CodebasePath -> Branch.Hash -> m (Maybe (Branch m))
+branchFromFiles cache rootDir h = time "FileCodebase.Common.branchFromFiles" $ do
   fileExists <- doesFileExist (branchPath rootDir h)
   if fileExists then Just <$>
-    Branch.read (deserializeRawBranch rootDir)
-                (deserializeEdits rootDir)
-                h
+    Branch.cachedRead
+      cache
+      (deserializeRawBranch rootDir)
+      (deserializeEdits rootDir)
+      h
```

Comment:
And here's where the cache finally gets used!

Reply:
Excite!!!!